src/components/PageDefinitions.js

```diff
@@ -181,62 +219,197 @@ class PageDefinitions extends Component {
+  }
+
+  provider = coordinates => {
+    return coordinates.provider ? coordinates.provider : null
+  }
+
+  type = coordinates => {
+    return coordinates.type ? coordinates.type : null
+  }
+
+  releaseDate = coordinates => {
+    const definition = this.props.definitions.entries[EntitySpec.fromCoordinates(coordinates).toPath()]
+    const described = get(definition, 'described')
+    if (described && described.releaseDate) return described.releaseDate
+    return null
+  }
+
+  license = coordinates => {
+    const definition = this.props.definitions.entries[EntitySpec.fromCoordinates(coordinates).toPath()]
+    const licensed = get(definition, 'licensed')
+    if (licensed && licensed.declared) return licensed.declared
+    return null
+  }
+
+  getSort(eventKey) {
+    switch (eventKey.value) {
+      case 'name':
+        return this.name
+      case 'namespace':
+        return this.namespace
+      case 'provider':
+        return this.provider
+      case 'type':
+        return this.type
+      case 'releaseDate':
+        return this.releaseDate
+      case 'license':
+        return this.license
+    }
+  }
+
+  onSort(eventKey) {
+    this.setState({ ...this.state, activeSort: eventKey.value, counter: this.state.counter + 1 })
+    this.props.dispatch(uiBrowseUpdateList({ sort: this.getSort(eventKey) }))
+  }
+
+  filterList(list) {
+    const { activeFilters } = this.state
+    if (activeFilters.length === 0) {
+      return list
+    }
+    return filter(list, component => {
+      const defintion = this.getDefinition(component)
+      for (let filterType in activeFilters) {
+        const value = activeFilters[filterType]
+        const fieldValue = this.getValue(defintion, filterType)
+        if (value === 'presence') {
+          if (!fieldValue) return false
+        } else if (value === 'absence') {
+          if (fieldValue) return false
+        } else {
+          if (!fieldValue || !fieldValue.toLowerCase().includes(value.toLowerCase())) {
+            return false
+          }
+        }
+      }
+      return true
+    })
+  }
+
+  onFilter(value) {
+    let activeFilters = Object.assign({}, this.state.activeFilters)
+    const filterValue = get(activeFilters, value.type)
+    if (filterValue && activeFilters[value.type] === value.value) delete activeFilters[value.type]
+    else activeFilters[value.type] = value.value
+    this.setState({ ...this.state, activeFilters })
+  }
+
+  incrementSequence() {
+    this.setState({ ...this.state, counter: this.state.counter + 1 })
   }
 
   noRowsRenderer() {
     return <div>Select components from the list above ...</div>
   }
 
-  renderContributeButton() {
+  checkSort(sortType) {
+    const { activeSort } = this.state
+    if (activeSort === sortType.value) return true
+    return false
+  }
+
+  checkFilter(filterType, id) {
+    const { activeFilters } = this.state
+    for (let filterIdx in activeFilters) {
```

Comment:
consider using `some`. here it seems like you want to know if the filter is active so any entry would make it happy.

Note also, avoid `==` in favor of `===`

Also, is `filterIdx` a number?  id? ... The name is a bit off. Feel free to spell out the whole word as well.